internal/grpc/grpclog/grpclog_test.go

```diff
@@ -0,0 +1,102 @@
+/*
+Copyright (c) Edgeless Systems GmbH
+
+SPDX-License-Identifier: AGPL-3.0-only
+*/
+
+// grpclog provides a logging utilities for gRPC.
+package grpclog
+
+import (
+	"context"
+	"fmt"
+	"sync"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"google.golang.org/grpc/connectivity"
+)
+
+func TestLogStateChanges(t *testing.T) {
+	testCases := []struct {
+		name    string
+		conn    getStater
+		isReady bool
+		assert  func(t *testing.T, lg *fakeLog, isReady bool)
+	}{
+		{
+			name: "log state changes",
+			conn: &fakeConn{
+				states: []connectivity.State{
+					connectivity.Connecting,
+					connectivity.Ready,
+				},
+			},
+			isReady: true,
+			assert: func(t *testing.T, lg *fakeLog, isReady bool) {
+				assert.Equal(t, "Connection state started as CONNECTING", lg.msgs[0])
+				assert.Equal(t, "Connection state changed to CONNECTING", lg.msgs[1])
+				assert.Equal(t, "Connection ready", lg.msgs[2])
+				assert.Len(t, lg.msgs, 3)
```

Comment:
This panics if the length isn't as expected. Check the length first using `require`.